app/models/piece.rb

```diff
@@ -7,7 +7,7 @@ def outside_board?(position_x, position_y)
   end
 
   def valid_move?(position_x, position_y)
-    if outside_board?(self.position_x, self.position_y)
+    if outside_board?(position_x, position_y) || is_obstructed()
       return false
     else
       return true
```

Comment:
if we return true here we'll never come to execute the logic on the next lines.. let's just drop the else statement.. meaning: if its outside board  the method returns false, otherwise it continues with the logic on the next lines..

so the first line can be just: `return false if outside_board?(position_x, position_y)...`

also: is_obsctructed() needs x and y arguments 😉